labman/db/plate.py

```diff
@@ -167,7 +167,12 @@ def search(samples=None, plate_notes=None, well_notes=None,
             else:
                 sql = sql_queries[0]
             TRN.add(sql, sql_args)
-            res = [Plate(pid) for pid in TRN.execute_fetchflatten()]
+
+            # explicitly sorting to ensure a deterministic result.
+            # sorting ids in python rather than with a SQL ORDER BY since
+            # there are several different SQL queries potentially being run.
+            sorted_pids = sorted(TRN.execute_fetchflatten())
+            res = [Plate(pid) for pid in sorted_pids]
```

Comment:
This is, I suspect, not an entirely straightforward question to answer because it is not immediately clear to me what the "obvious" value to sort by would be.  I chose to sort by the db id for the plate because it is easy and well-behaved and I honestly don't care what order they come back in as long as it's deterministic :)  However, plates also hold a bunch of other fields that would be reasonable to sort by (e.g., what the user actually calls them (external_id) and create date, and so on).  In the interface, we currently let users sort by at least two of these already, figuring they might want different ones at different times.